exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillStatementImpl.java

```diff
@@ -38,8 +44,12 @@
 // methods for compatibility.)
 class DrillStatementImpl extends AvaticaStatement implements DrillStatement,
                                                              DrillRemoteStatement {
+  //Not using the DrillbitContext's ExecutorService as this is threadPool is light-weight (threads wake up to cancel tasks) but needs a low response time
+  private static ExecutorService queryTimeoutTaskPool = Executors.newCachedThreadPool(new NamedThreadFactory("q-timeout-"));
```

Comment:
Maybe simpler, but I'm doubtful this is less overhead: there's a new executor and a new thread created for each statement (to compare with just telling the system how long to wait in DrillCursor, knowing that the i/o system is using a fixed number of threads vs your proposal which is unbounded). 
I'm also not following why you would need a new state, as this would be a pure client thing.